packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -302,8 +302,17 @@ const defaultTwoColumnModuleLayout = <T: { +[string]: mixed }>({
     if (hasTwoColumnItems) {
       // If the number of items to position is greater that the batch size
       // we identify the batch with the two column item and apply the graph only to those items
-      const splitIndex = calculateSplitIndex(itemsWithoutPositions);
-      const shouldBatchItems = itemsWithoutPositions.length > TWO_COL_ITEMS_MEASURE_BATCH_SIZE;
+      const twoColumnIndex = itemsWithoutPositions.indexOf(twoColumnItems[0]);
+
+      const skipGraph =
+        heights.every((height) => height === 0) && twoColumnIndex < heights.length - 1;
```

Comment:
more of an optimization but i don't think height specifically needs to be 0 here (e.g. we could say something like skip graph and just place where it is if the deviation of the heights in the array is less than the size of the gutter)

okay as a followup though. also, i don't think we need the check for twoColumnIndex < heights.lenght - 1?

Reply:
Yeah, I was thinking on how to apply the rule on a more general way but this is more accurate on what we want to solve at the moment. I can create a ticket to follow this as an algorithm improvement.

Reply:
okay that sounds good - can we leave a comment here so we don't lose track of this? also, can you confirm we should be able to remove the `twoColumnIndex < heights.lenght - 1`

Reply:
We can remove it, the side effect is that the item is going to be positioned as the next item always without calculating a better position, tbh that check doesn't solve that either so we could remove it. We would have to implement something to better position on that case.